README.md

```diff
@@ -1,8 +1,362 @@
+For more information about our decision to unify these projects, see our announcement here: (TODO)
+
+## Overview
+
+The concept for the CUDA C++ Core Libraries (CCCL) grew organically out of the Thrust, CUB, and libcudacxx projects that were developed independently over the years with a similar goal: to provide a high-quality, high-performance, and easy-to-use C++ abstractions for CUDA developers.
+Naturally, there was a lot of overlap among the three projects, and we realized that we could better serve the community by unifying them into a single repository.
+
+- **Thrust** is the C++ parallel algorithms library which inspired the introduction of parallel algorithms to the C++ Standard Library. Thrust's high-level interface greatly enhances programmer productivity while enabling performance portability between GPUs and multicore CPUs via configurable backends that allow using multiple parallel programming frameworks (such as CUDA, TBB, and OpenMP).
+
+- **libcudacxx** is the CUDA C++ Standard Library. It provides an implementation of the C++ Standard Library that works in both host and device code. Additionally, it provides abstractions for CUDA-specific hardware features like synchronization primitives, cache control, atomics, and more. 
+
+- **CUB** is a lower-level, CUDA-specific library designed for speed-of-light parallel algorithms across all GPU architectures. In addition to device-wide algorithms, it provides *cooperative algorithms* like block-wide reduce and warp-wide scan, providing CUDA kernel developers with building blocks to create speed-of-light, custom kernels. 
+
+Bringing these libraries together enables us to better deliver the features you need as well as provide a more cohesive experience where everything “just works” out of the box.  
+Our long-term goal is to offer a single, unified, modern C++ library that merges and extends the functionality of our existing libraries and more. 
+We envision the CUDA C++ Core Libraries to fill a similar role that the Standard C++ Library fills for Standard C++ – equipping you with general-purpose, speed-of-light tools to focus on solving the problems you care about, not boilerplate.
+
+## Example
+
+This is a simple example demonstrating the use of CCCL functionality from Thrust, CUB, and libcudacxx.
+
+It shows how to use Thrust/CUB/libcudacxx to implement a simple parallel reduction kernel. Each thread block
+computes the sum of a subset of the array using `cub::BlockRecuce`. The sum of each block is then reduced 
+to a single value using an atomic add via `cuda::atomic_ref` from libcudacxx.
+
+It then shows how the same reduction can be done using Thrust's `reduce` algorithm and compares the results.
+
+```cpp
+#include <thrust/device_vector.h>
+#include <cstdio>
+#include <cub/block/block_reduce.cuh>
+#include <cuda/atomic>
```

Comment:
This seems like a random include order. I don't know what CCCL wants to recommend for style here, but maybe don't mix STL and CCCL includes?